src/core/shader.js

```diff
@@ -0,0 +1,156 @@
+var debug = require('../utils/debug');
+var schema = require('./schema');
+var error = debug('core:register-shader:error');
+
+var processSchema = schema.process;
+var shaders = module.exports.shaders = {};  // Keep track of registered shaders.
```

Comment:
There's redundancy with `shaders` and `shaderList`, and at first wasn't sure which was which.